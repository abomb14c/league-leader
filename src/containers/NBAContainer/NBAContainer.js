import React from 'react';
import {connect} from 'react-redux';
import NBACard from '../../components/NBACard/NBACard';

export const NBAContainer = ({ NBA }) => {
  
  const eastTeams = NBA['EASTERN CONFERENCE'].map( team => {
    return <NBACard {...team} />;
  });

  const westTeams = NBA['WESTERN CONFERENCE'].map( team => {
    return <NBACard {...team} />;
  });

  return (
    <div>
      {eastTeams}
      {westTeams}
    </div>
  );
};

export const mapStateToProps = state => ({
  NBA: state.NBA
});

export default connect(mapStateToProps, null)(NBAContainer);