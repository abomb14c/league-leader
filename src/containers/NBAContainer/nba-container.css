.nba-container {
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.30);
  display: flex;
  flex-direction: column;
  height: 1460px;
  margin-left: 5%;
  margin-top: 3%;
  width: 40%;
}

.nba-header {
  background-color: #3B3738;
  border-bottom: 2px solid #C63D0F;
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.30);
  color: #FDF3E7;
  display: flex;
  justify-content: space-between;
  font-weight: 200;
  padding-bottom: 5px;
  padding-right: 10%;
}

.conference-title-container {
  align-items: center;
  border-top: 1px solid #C63D0F;
  background-color: #3B3738;
  display: flex;
  height: 50px;
  padding-left: 2%;
}

.conference-name {
  margin: none;
  color: #FDF3E7;
}