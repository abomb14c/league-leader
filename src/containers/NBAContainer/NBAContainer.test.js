import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import  { NBAContainer, mapStateToProps}  from './NBAContainer';

describe('NBAContainer', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {

    mockProps = {
      NBA: {'EASTERN CONFERENCE':[{}, {}, {}],
        'WESTERN CONFERENCE': [{}, {}, {}]
      }
    };

    wrapper = shallow(<NBAContainer
      {...mockProps}
    />);
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should map the NBA to props', () => {
    const mockState = {
      league: {},
      NBA: {'EASTERN CONFERENCE':[{}, {}, {}],
        'WESTERN CONFERENCE': [{}, {}, {}]
      },
      draftTeams: [{}, {}, {}]
    };

    const expected = {
      NBA: {'EASTERN CONFERENCE':[{}, {}, {}],
        'WESTERN CONFERENCE': [{}, {}, {}]
      }
    };
    
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
