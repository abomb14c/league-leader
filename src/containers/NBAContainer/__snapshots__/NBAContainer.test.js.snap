// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NBAContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    NBA={
        Object {
            "EASTERN CONFERENCE": Array [
              Object {},
              Object {},
              Object {},
            ],
            "WESTERN CONFERENCE": Array [
              Object {},
              Object {},
              Object {},
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="nba-header"
>
          <h3
                    className="header-desc"
          >
                    Name
          </h3>
          <h3
                    className="header-desc nba-record"
          >
                    Record
          </h3>
</div>,
        <div
          className="conference-container"
>
          <div
                    className="conference-title-container"
          >
                    <h3
                              className="conference-name"
                    >
                              Eastern Conference
                    </h3>
          </div>
          <NBACard />
          <NBACard />
          <NBACard />
</div>,
        <div>
          <div
                    className="conference-container"
          >
                    <div
                              className="conference-title-container"
                    >
                              <h3
                                        className="conference-name"
                              >
                                        Western Conference
                              </h3>
                    </div>
          </div>
          <NBACard />
          <NBACard />
          <NBACard />
</div>,
      ],
      "className": "nba-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="header-desc"
>
              Name
</h3>,
            <h3
              className="header-desc nba-record"
>
              Record
</h3>,
          ],
          "className": "nba-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name",
              "className": "header-desc",
            },
            "ref": null,
            "rendered": "Name",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Record",
              "className": "header-desc nba-record",
            },
            "ref": null,
            "rendered": "Record",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="conference-title-container"
>
              <h3
                            className="conference-name"
              >
                            Eastern Conference
              </h3>
</div>,
            Array [
              <NBACard />,
              <NBACard />,
              <NBACard />,
            ],
          ],
          "className": "conference-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3
                className="conference-name"
>
                Eastern Conference
</h3>,
              "className": "conference-title-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Eastern Conference",
                "className": "conference-name",
              },
              "ref": null,
              "rendered": "Eastern Conference",
              "type": "h3",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="conference-container"
>
              <div
                            className="conference-title-container"
              >
                            <h3
                                          className="conference-name"
                            >
                                          Western Conference
                            </h3>
              </div>
</div>,
            Array [
              <NBACard />,
              <NBACard />,
              <NBACard />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="conference-title-container"
>
                <h3
                                className="conference-name"
                >
                                Western Conference
                </h3>
</div>,
              "className": "conference-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3
                  className="conference-name"
>
                  Western Conference
</h3>,
                "className": "conference-title-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Western Conference",
                  "className": "conference-name",
                },
                "ref": null,
                "rendered": "Western Conference",
                "type": "h3",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="nba-header"
>
            <h3
                        className="header-desc"
            >
                        Name
            </h3>
            <h3
                        className="header-desc nba-record"
            >
                        Record
            </h3>
</div>,
          <div
            className="conference-container"
>
            <div
                        className="conference-title-container"
            >
                        <h3
                                    className="conference-name"
                        >
                                    Eastern Conference
                        </h3>
            </div>
            <NBACard />
            <NBACard />
            <NBACard />
</div>,
          <div>
            <div
                        className="conference-container"
            >
                        <div
                                    className="conference-title-container"
                        >
                                    <h3
                                                className="conference-name"
                                    >
                                                Western Conference
                                    </h3>
                        </div>
            </div>
            <NBACard />
            <NBACard />
            <NBACard />
</div>,
        ],
        "className": "nba-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="header-desc"
>
                Name
</h3>,
              <h3
                className="header-desc nba-record"
>
                Record
</h3>,
            ],
            "className": "nba-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name",
                "className": "header-desc",
              },
              "ref": null,
              "rendered": "Name",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Record",
                "className": "header-desc nba-record",
              },
              "ref": null,
              "rendered": "Record",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="conference-title-container"
>
                <h3
                                className="conference-name"
                >
                                Eastern Conference
                </h3>
</div>,
              Array [
                <NBACard />,
                <NBACard />,
                <NBACard />,
              ],
            ],
            "className": "conference-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h3
                  className="conference-name"
>
                  Eastern Conference
</h3>,
                "className": "conference-title-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Eastern Conference",
                  "className": "conference-name",
                },
                "ref": null,
                "rendered": "Eastern Conference",
                "type": "h3",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="conference-container"
>
                <div
                                className="conference-title-container"
                >
                                <h3
                                                className="conference-name"
                                >
                                                Western Conference
                                </h3>
                </div>
</div>,
              Array [
                <NBACard />,
                <NBACard />,
                <NBACard />,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="conference-title-container"
>
                  <h3
                                    className="conference-name"
                  >
                                    Western Conference
                  </h3>
</div>,
                "className": "conference-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h3
                    className="conference-name"
>
                    Western Conference
</h3>,
                  "className": "conference-title-container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Western Conference",
                    "className": "conference-name",
                  },
                  "ref": null,
                  "rendered": "Western Conference",
                  "type": "h3",
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
